web-ui/src/connectors/DataSourceTable.tsx

```diff
@@ -0,0 +1,171 @@
+// The table that displays the connectors.
+//
+// Table allows to edit the name and description in the table directly and can
+// delete/edit individual connectors.
+
+import { useState, useCallback } from 'react'
+import Box from '@mui/material/Box'
+import Card from '@mui/material/Card'
+import Typography from '@mui/material/Typography'
+import CustomChip from 'src/@core/components/mui/chip'
+import { GridColumns, GridRenderCellParams, useGridApiRef } from '@mui/x-data-grid-pro'
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { ConnectorService } from 'src/types/manager/services/ConnectorService'
+import { ConnectorDescr } from 'src/types/manager/models/ConnectorDescr'
+import { CancelError, UpdateConnectorRequest, UpdateConnectorResponse } from 'src/types/manager'
+import EntityTable from 'src/components/table/EntityTable'
+import useStatusNotification from 'src/components/errors/useStatusNotification'
+import { ConnectorDialog, getStatusObj } from 'src/types/connectors'
+
+const DataSourceTable = () => {
+  const [rows, setRows] = useState<ConnectorDescr[]>([])
+  const { pushMessage } = useStatusNotification()
+  const apiRef = useGridApiRef()
+  const queryClient = useQueryClient()
+
+  // Query to retrieve table content
+  const fetchQuery = useQuery<ConnectorDescr[]>(['connector'])
+
+  // Update row name and description if edited in the cells:
+  const mutation = useMutation<UpdateConnectorResponse, CancelError, UpdateConnectorRequest>(
+    ConnectorService.updateConnector
+  )
+  const processRowUpdate = useCallback(
+    (newRow: ConnectorDescr, oldRow: ConnectorDescr) => {
+      mutation.mutate(
+        {
+          connector_id: newRow.connector_id,
+          description: newRow.description,
+          name: newRow.name
+        },
+        {
+          onSettled: () => {
+            queryClient.invalidateQueries(['connector'])
+            queryClient.invalidateQueries(['connectorStatus', { connector_id: newRow.connector_id }])
+          },
+          onError: (error: CancelError) => {
+            pushMessage({ message: error.message, key: new Date().getTime(), color: 'error' })
+            apiRef.current.updateRows([oldRow])
+          }
+        }
+      )
+
+      return newRow
+    },
+    [apiRef, mutation, pushMessage, queryClient]
+  )
+
+  // Delete a connector entry
+  const deleteMutation = useMutation<void, CancelError, number>(ConnectorService.deleteConnector)
+  const deleteSource = useCallback(
+    (cur_row: ConnectorDescr) => {
+      setTimeout(() => {
+        const oldRow = rows.find(row => row.connector_id === cur_row.connector_id)
+        if (oldRow !== undefined) {
+          deleteMutation.mutate(cur_row.connector_id, {
+            onSettled: () => {
+              queryClient.invalidateQueries(['connector'])
+            },
+            onSuccess: () => {
+              setRows(prevRows => prevRows.filter(row => row.connector_id !== cur_row.connector_id))
```

Comment:
That's what the `queryClient.invalidateQueries(['connector'])` will eventually do when it re-renders. But you want the state that's most likely to show up show up immediately and not a second later otherwise it's confusing for the user if it flickers form old to new.